test/cgroups.bats

```diff
@@ -10,19 +10,47 @@ function teardown() {
 	cleanup_test
 }
 
-@test "pids limit" {
-	if ! grep -qEw ^pids /proc/cgroups; then
-		skip "pids cgroup controller is not available"
-	fi
-	CONTAINER_PIDS_LIMIT=1234 start_crio
+function run_pids_limit_test() {
+	local set_limit="$1"
+	local expected_limit="$2"
 
+	CONTAINER_PIDS_LIMIT="$set_limit" start_crio
 	jq '	  .command'='["/bin/sleep", "600"]' \
 		"$TESTDATA"/container_config.json > "$TESTDIR"/container_pids_limit.json
 
 	ctr_id=$(crictl run "$TESTDIR"/container_pids_limit.json "$TESTDATA"/sandbox_config.json)
 
 	output=$(crictl exec --sync "$ctr_id" sh -c 'cat /sys/fs/cgroup/pids/pids.max 2>/dev/null || cat /sys/fs/cgroup/pids.max')
-	[[ "$output" == "1234" ]]
+	[[ "$output" == "$expected_limit" ]]
+}
+
+
+@test "pids limit" {
+	if ! grep -qEw ^pids /proc/cgroups; then
+		skip "pids cgroup controller is not available"
+	fi
+	run_pids_limit_test "1234" "1234"
+}
+
+@test "negative pids limit" {
+	if ! grep -qEw ^pids /proc/cgroups; then
+		skip "pids cgroup controller is not available"
+	fi
+	run_pids_limit_test "-1" "max"
+}
+
+@test "zero pids limit" {
+	if ! grep -qEw ^pids /proc/cgroups; then
+		skip "pids cgroup controller is not available"
+	fi
+	pids_root=/sys/fs/cgroup/pids
+	if is_cgroup_v2; then
+		pids_root=/sys/fs/cgroup
+	fi
+
+	# pids_limit of 0 sets the pids.max to that of pid 1
+	parent_limit=$(cat "$pids_root/$(grep pid '/proc/1/cgroup' | tr ':' ' ' | awk '{ printf $3 }')/pids.max")
```

Comment:
1. This won't work for cgroup v2 as /proc/$PID/cgroup looks like:
```
kir@kir-rhat sys]$ cat /proc/1/cgroup 
0::/init.scope
[kir@kir-rhat sys]$ cat /proc/self/cgroup 
0::/user.slice/user-1000.slice/user@1000.service/vte-spawn-d48a4976-89a3-4aac-96b6-b783b46a6113.scope
```

2. `awk -F : '$2 == "pid" {print $3}' /proc/1/cgroup` does the same as `grep pid '/proc/1/cgroup' | tr ':' ' ' | awk '{ printf $3 }'`

Reply:
And to amend this for cgroupv2 case (not tested)

```sh
	pids_root=/sys/fs/cgroup/pids
	lookfor=pids
￼	if is_cgroup_v2; then
￼		pids_root=/sys/fs/cgroup
		lookfor=""
￼	fi
	path=$pids_root/$(awk -v lookfor="$lookfor" -F : '$2 == lookfor {print $3; exit}' /proc/self/cgroup)/pids.max
```

Reply:
updated!